Automattic/wp-calypso
# Associate a site with a GMB location

Requires server side patch D11380.

This PR update the GMB service in sharing to allow users to pick a GMB location and associate this location with the current site.

<img width="624" alt="screenshot" src="https://user-images.githubusercontent.com/594356/38485079-ebc507d0-3bd8-11e8-98f5-bff7c97ed54e.png">

This is done by adding 2 new properties to the siteSettings:
- `google_my_business_keyring_id`
- `google_my_business_location_id `

I think we should review the general design/idea before proceeding further. The drawback of this approach is that it will change what it means to be connected to Google My Business: authorizing WordPress.com (through the OAuth2 flow) is not enough you now need to pick a location or it won't be marked as connected in the sharing page.

### Testing Instructions
- Load this branch locally
- Go to the [`Sharing` page](http://calypso.localhost:3000/sharing)
- Try Connecting with Google My Business
- Pick a location, ensure it is displayed when you expand the card
- Disconnect
- Reconnect
- Try connecting with the same location from another site

### Reviews
